Cleaned up user-facing configuration options.

Aaronius opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

Removed configuration options that will not be used. Moved configuration options used for testing out of the sandbox and into the library itself, so that it's not consumer-facing.

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context
Don't confuse users with configuration options that don't apply to them.
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

liljenback commented:
You could just add these as temp defaults now that the configuration code has been merged. But why not keep the test config separate from the code base?

Aaronius commented:
I've rebased on master and moved the values to validator defaults. @liljenback, the reason we're moving these options out of the sandbox is because we don't want them to keep showing up in demos. They only mean something to the engineering team and were confusing other people. We discussed this in yesterday's sprint planning. Let us know if you have a suggestion of a better place to put them that would resolve the concerns.
